sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -312,44 +310,53 @@ class AllCombineFn(AccumulatorCombineFn):
   _accumulator_type = AllAccumulator
 
 
-MAX_LONG_10_FOR_LEADING_ZEROS = [19, 18, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15,
-                                 15, 15, 15, 14, 14, 14, 13, 13, 13, 12, 12, 12,
-                                 12, 11, 11, 11, 10, 10, 10, 9, 9, 9, 9, 8, 8,
-                                 8, 7, 7, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 3, 3,
-                                 3, 3, 2, 2, 2, 1, 1, 1, 0, 0, 0]
-
-LONG_SIZE = 64
-
-
-def compare_to(x, y):
-  """return the sign bit of x-y"""
-  if x < y:
-    return 1
-  return 0
+POWER_TEN = [10e-1, 10e0, 10e1, 10e2, 10e3, 10e4, 10e5,
+             10e6, 10e7, 10e8, 10e9, 10e10, 10e11,
+             10e12, 10e13, 10e14, 10e15, 10e16, 10e17,
+             10e18]
 
 
 def get_log10_round_to_floor(element):
-  number_of_leading_zeros = LONG_SIZE - element.bit_length()
-  y = MAX_LONG_10_FOR_LEADING_ZEROS[number_of_leading_zeros]
-  return y - compare_to(element, math.pow(10, y))
+  power = 0
+  while element >= POWER_TEN[power]:
+    power += 1
+  return power - 1
+
+
+class DataflowDistributionCounter(object):
+  """Pure python DataflowDistributionCounter in case Cython not available.
```

Comment:
<!--new_thread; commit:2e80f6f09baf2949f8b46af1b2858a2fa6ee567b; resolved:0-->
Can we rename this to `SlowDataflowDistributionCounter`?  Then, we can do:

```python
try:
  from apache_beam.transforms.cy_dataflow_distribution_counter import DataflowDistributionCounter
except:
  DistributionCounter = SlowDataflowDistributionCounter
```

Below, we can get rid of the `try` / `except` in `DataflowDistributionCounterFn` and just do `_accumulator_type = DataflowDistributionCounter`.

Further, explicitly exposing the slow counter lets us test both in tests (see the test file).

Reply:
With current pattern in ```transforms/__init__.py```, we can do the same thing as you mentioned without renaming. For test both, I had more explainations below.

Reply:
and for here, I feel confused about where you suggested to put this pattern?
```
try:
  from apache_beam.transforms.cy_dataflow_distribution_counter import DataflowDistributionCounter
except:
  DistributionCounter = SlowDataflowDistributionCounter
```

Reply:
You can have that in the file directly, right after the class definitionfor example:

```python
class SlowDataflowDistributionCounter:
   ...

try:
  from apache_beam.transforms.cy_dataflow_distribution_counter import DataflowDistributionCounter
except:
  DistributionCounter = SlowDataflowDistributionCounter
```

Thank you for pointing out that we run both python-only and cython tests.  Please ignore that comment in this case.